shell/browser/api/atom_api_menu.cc

```diff
@@ -14,6 +16,13 @@
 #include "shell/common/native_mate_converters/string16_converter.h"
 #include "shell/common/node_includes.h"
 
+namespace {
+// We need this map to keep references to currently opened menus.
+// Without this menus would be destroyed by js garbage collector
+// even when they are still displayed.
+std::map<uint32_t, v8::Global<v8::Object>> g_menus;
```

Comment:
Maybe it's because it's early in the morning, but I'm not 100% convinced this will free the memory.  cc @deepak1556 will this leak the objects?

Reply:
This would have leaked, but its being manually tracked in menu creation and close. So its fine.